backend/cms/models/event.py

```diff
@@ -0,0 +1,69 @@
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.db import models
+
+from cms.models import Site
+from cms.models.poi import POI
+
+
+class Event(models.Model):
+    site = models.ForeignKey(Site)
+    location = models.ForeignKey(POI, on_delete=models.PROTECT, null=True, blank=True)
+    start_date = models.DateTimeField()
+    end_date = models.DateTimeField()
+    picture = models.ImageField(null=True, blank=True, upload_to='events/%Y/%m/%d')
+    is_all_day = models.BooleanField(default=False)
+    is_recurring = models.BooleanField(default=False)
+    has_recurring_end_date = models.BooleanField(default=False)
+    recurring_end_date = models.DateField(null=True, default=None, blank=True)
+    FREQUENCY = (
+        ('daily', 'Täglich'),
+        ('weekly', 'Wöchentlich'),
+        ('monthly', 'Monatlich'),
+        ('yearly', 'Jährlich')
+    )
+    frequency = models.CharField(max_length=7, choices=FREQUENCY, null=True, blank=True, default=None)
+
+    def clean(self):
```

Comment:
Just general note without need to change: I think "validate" is a better name for such functions? "clean" could implicate an state changing clean up method.

Reply:
That's a Django built-in function: https://docs.djangoproject.com/en/2.1/ref/models/instances/#validating-objects